blueprints/ember-bootstrap/index.js

```diff
@@ -6,60 +6,151 @@ const fs = require('fs');
+  beforeInstall(option) {
+    let { bootstrapVersion, targetPreprocessor } = option;
+    if (bootstrapVersion !== 3 && bootstrapVersion !== 4) {
+      throw new SilentError('Bootstrap version must be 3 or 4');
+    }
+
+    if (targetPreprocessor && !validPreprocessors.includes(targetPreprocessor)) {
+      throw new SilentError(`Valid preprocessors are: ${validPreprocessors.join(', ')}`);
+    }
+
+    if (!targetPreprocessor) {
+      let dependencies = this.project.dependencies();
+      if ('ember-cli-sass' in dependencies) {
+        targetPreprocessor = option.targetPreprocessor = 'sass';
+      } else if ('ember-cli-less' in dependencies) {
+        targetPreprocessor = option.targetPreprocessor = 'less';
+      } else {
+        targetPreprocessor = 'none';
+      }
+    }
+
+    if (bootstrapVersion === 4 && targetPreprocessor === 'less') {
+      throw new SilentError('You cannot use LESS with Bootstrap 4');
+    }
+
+    this.ui.writeLine(chalk.green(`Installing for Bootstrap ${bootstrapVersion} using preprocessor ${targetPreprocessor}`));
+  },
+
+  afterInstall(option) {
+    return this.adjustBootstrapDependencies(option)
+      .then(() => this.addPreprocessorStyleImport(option))
+      .then(() => this.addBuildConfiguration(option));
   },
 
-  addDependencies() {
+  adjustBootstrapDependencies(option) {
+    let { bootstrapVersion, targetPreprocessor } = option;
     let dependencies = this.project.dependencies();
+    let bowerDependencies = this.project.bowerDependencies();
+    let promises = [];
 
-    let promises = [
-      this.addPackageToProject('bootstrap', bs4Version)
-    ];
-
-    if ('ember-cli-sass' in dependencies) {
+    if (bootstrapVersion === 4) {
+      if ('bootstrap' in bowerDependencies) {
+        this.ui.writeLine(chalk.white.bgBlue('ACTION: Manually remove \'bootstrap\' from bower.json'));
+      }
+      if ('bootstrap-sass' in dependencies) {
+        promises.push(this.removePackageFromProject('bootstrap-sass'));
+      }
+      if ('bootstrap-sass' in bowerDependencies) {
+        this.ui.writeLine(chalk.white.bgBlue('ACTION: Manually remove \'bootstrap-sass\' from bower.json'));
+      }
+      promises.push(this.addPackageToProject('bootstrap', bs4Version));
+    } else if (targetPreprocessor === 'sass') {
+      if ('bootstrap' in dependencies) {
+        promises.push(this.removePackageFromProject('bootstrap'));
+      }
+      if ('bootstrap' in bowerDependencies) {
+        this.ui.writeLine(chalk.white.bgBlue('ACTION: Manually remove \'bootstrap\' from bower.json'));
+      }
       promises.push(this.addPackageToProject('bootstrap-sass', bs3Version));
     } else {
+      if ('bootstrap' in dependencies) {
+        promises.push(this.removePackageFromProject('bootstrap'));
+      }
+      if ('bootstrap-sass' in dependencies) {
+        promises.push(this.removePackageFromProject('bootstrap-sass'));
+      }
       promises.push(this.addBowerPackageToProject('bootstrap', bs3Version));
```

Comment:
I agree. That's one of the things I wanted to discuss with you. I'll make that change.